frontend/packages/pydantic-forms/src/core/helper.ts

```diff
@@ -51,27 +51,83 @@ export const getErrorDetailsFromResponse = function (
 };
 
 /**
- * Easy pluk options or format from the anyOf, allOf or oneOf keys
- *
- * @param field A field from the 'properties' key of the JSON Schema
- * @returns anyOf, allOf, or allOf value
+ * This functions looks to see if the field has any of the anyOf, allOf or oneOf keys. If it
+ * does hoists these properties to the root level. This is for use cases where we get field definitions
+ * like this:
+ * {
+ *     "id": "bfd",
+ *    "type": "object",
+ *    "allOf": [
+ *        {... propertySchemaParsed}
+ *        {type: "null"}
+ *     ]
+ * }
  */
-export const getFieldAllOfAnyOfEntry = (
+export const flattenSchemaCombinators = (
     propertySchemaParsed: PydanticFormPropertySchemaParsed,
-) => {
-    const optionFields = [
-        propertySchemaParsed.anyOf,
-        propertySchemaParsed.oneOf,
-        propertySchemaParsed.allOf,
-    ];
-
-    for (const optionsDefs of optionFields) {
-        if (!optionsDefs) {
-            continue;
+): PydanticFormPropertySchemaParsed => {
+    if (propertySchemaParsed.allOf) {
+        const allOfProperties = propertySchemaParsed.allOf.reduce(
+            (mergedItem, allOfItem) => {
+                return { ...mergedItem, ...allOfItem };
+            },
+        );
+
+        if (propertySchemaParsed.anyOf || propertySchemaParsed.oneOf) {
+            console.warn(
+                'Pydantic forms does not currently support combining allOf with anyOf or oneOf. This may lead to unexpected behavior.',
+            );
         }
 
-        return optionsDefs;
+        return {
+            ...allOfProperties,
+            ...propertySchemaParsed,
+        };
     }
+
+    if (propertySchemaParsed.oneOf) {
+        if (propertySchemaParsed.anyOf) {
+            console.warn(
+                'Pydantic forms does not support combining oneOf with anyOf. This may lead to unexpected behavior.',
+            );
+        }
+
+        if (
+            (isNullable(propertySchemaParsed) &&
+                propertySchemaParsed.oneOf.length > 2) ||
+            (!isNullable(propertySchemaParsed) &&
+                propertySchemaParsed.oneOf.length > 1)
```

Comment:
Even though this should be impossible, the `isNullable` may also return true if it finds a null in `propertySchemaParsed.allOf`. Maybe only check if `propertySchemaParsed.oneOf` is nullable?